### ISSUE TITLE
`firebase deploy --only <targets>` deploying wrong targets

### ISSUE THREAD

**Issue by kenhoff** [firebase/firebase-tools#180]
Hey folks - when I try to deploy my dev environment using `firebase deploy --only default`, it deploys to my production environment instead. Yikes!

Not sure what's going on here. I've created two firebases, `exampleapp-dev` and `exampleapp-prod`. 

Here's what my `.firebaserc` looks like:

```
{
  "projects": {
    "default": "exampleapp-dev",
    "prod": "exampleapp-prod"
  }
}
```

Here's the output of running `firebase deploy --only default`:

```
ken@isaac ~/d/firebase-tools-bug> firebase deploy --only default

=== Deploying to 'exampleapp-prod'...

i  deploying
i  starting release process (may take several minutes)...

✔  Deploy complete!

Dashboard: https://console.firebase.google.com/project/exampleapp-prod/overview
```

I'm using `firebase-tools` version `3.0.4` and fish `2.2.0`, but works the same in bash.

Any thoughts?

**Comment by mbleigh** [comment]
`firebase deploy --only [features]` is meant for selecting which **features** you want to deploy, not to which **project**. So you might do `firebase deploy --only hosting`.

What you're looking for is `firebase deploy --project default`. I'll take this as a sign that we could use better documentation in this area.

**Comment by kenhoff** [comment]
Ah, got it! Yeah, I didn't see the `--project` option in the CLI help. Thanks!

**Comment by dzoba** [comment]
I just ran into the same issue.  Thanks @kenhoff for reporting this before me!  yes I agree, could definitely use some more documentation.  The CLI doesn't mention the --project option

![image](https://cloud.githubusercontent.com/assets/857664/17157370/a9cef69e-5343-11e6-9515-252221b57668.png)

**Comment by nickwph** [comment]
You can also do this to set the default project to use
```
firebase use <project>
```
So the following will deploy to development
```
firebase use development
firebase deploy
```

**Comment by JasonGoemaat** [comment]
Thanks!  For me I have a couple firebase projects and it kept deploying to one project (call it 'dumbtest-50234'), even though that was not specified *anywhere* in my project, .firebaserc, firebase.json, etc.  When I used 'firebase init' I specified another project completely (call it 'dumbtest-test') and I could not find anywhere that the other project was specified.  I had to do a 'firebase use dumbtest-test' and then it deployed to the correct project.  I guess firebase version 3.9.1 completely ignores the project in your config and stores what project you are using somewhere globally?
